huaweicloud/services/acceptance/eip/resource_huaweicloud_vpc_bandwidth_test.go

```diff
@@ -168,6 +168,51 @@ func TestAccVpcBandWidth_prePaid(t *testing.T) {
 	})
 }
 
+func TestAccVpcBandWidth_changeToPeriod(t *testing.T) {
+	var bandwidth bandwidths.BandWidth
+
+	randName := acceptance.RandomAccResourceName()
+	resourceName := "huaweicloud_vpc_bandwidth.test"
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&bandwidth,
+		getBandwidthResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckChargingMode(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVpcBandWidth_basic(randName, 5),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", randName),
+				),
+			},
+			{
+				Config: testAccVpcBandWidth_prePaid(randName, 5, false),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "charging_mode", "prePaid"),
+					resource.TestCheckResourceAttr(resourceName, "auto_renew", "false"),
+				),
+			},
+			{
+				Config: testAccVpcBandWidth_prePaid(randName, 5, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "charging_mode", "prePaid"),
+					resource.TestCheckResourceAttr(resourceName, "auto_renew", "true"),
+				),
+			},
```

Comment:
you can add an expect error check
```suggestion
			},
			{
				Config:      testAccVpcBandWidth_basic(randName, 5),
				ExpectError: regexp.MustCompile(`xxxx`),
				),
			},
```